recipes/powershell5.rb

```diff
@@ -31,11 +33,19 @@
 
   if windows_version.windows_server_2012_r2? || windows_version.windows_8_1?
 
+    powershell_version_cmd = 'powershell.exe $PSVersionTable.PSVersion.Major'
+    shell_out = Mixlib::ShellOut.new(powershell_version_cmd)
+    shell_out.run_command
+    powershell_version = shell_out.stdout.to_f
+
+    # Reboot if user specifies immediate_reboot
+    include_recipe 'powershell::windows_reboot' if node['powershell']['installation_reboot_mode'] == 'immediate_reboot' && powershell_version < 5
+
     windows_package 'Windows Management Framework Core 5.0' do
       source node['powershell']['powershell5']['url']
       checksum node['powershell']['powershell5']['checksum']
       installer_type :custom
-      options '/quiet /norestart'
+      options '/quiet'
       action :install
       not_if { registry_data_exists?('HKLM\SOFTWARE\Microsoft\PowerShell\3\PowerShellEngine', { :name => 'PowerShellVersion', :type => :string, :data => '5.0.9701.0' }) }
```

Comment:
@adamedx , this is correct. There are only two folders that exist in registery: `\\powershell\\1` and `\\powershell\\3`. Registery data for powershell2 is stored in `\\powershell\\1` and for powershell3, powershell4 and powershell5, it's stored in `\\powershell\\3`. 
They don't depict the powershell version. Version is depicted by the `data` field. e.g. in case of powershell5, its `:data => '5.0.9701.0'`